libuavcan/include/libuavcan/transport/media/channels.hpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ */
+/** @file */
+
+#ifndef LIBUAVCAN_TRANSPORT_MEDIA_CHANNELS_HPP_INCLUDED
+#define LIBUAVCAN_TRANSPORT_MEDIA_CHANNELS_HPP_INCLUDED
+
+#include <array>
+#include <algorithm>
+
+#include "libuavcan/libuavcan.hpp"
+#include "libuavcan/introspection.hpp"
+#include "libuavcan/time.hpp"
+
+namespace libuavcan
+{
+/**
+ * @namespace transport
+ * Contains transport-specific types and namespaces.
+ */
+namespace transport
+{
+/**
+ * @namespace media
+ * Contains media-specific types and namespaces.
+ *
+ * See the @ref MediaDevGuide for details
+ * on porting the media layer to a given platform.
+ */
+namespace media
+{
+/**
+ * Filter config struct.
+ * TODO: document and examples
+ */
+struct LIBUAVCAN_EXPORT FilterConfig
+{
+    std::uint32_t id;
+    std::uint32_t mask;
+
+    bool operator==(const FilterConfig& rhs) const
+    {
+        return rhs.id == id && rhs.mask == mask;
+    }
+
+    FilterConfig()
+        : id(0)
+        , mask(0)
```

Comment:
?

Reply:
Sorry, I should have been more verbose. What I was trying to communicate is that we should use field initializers instead of ctor init list here, per the rules we've discussed earlier.